chore: update showcase version for sequence tests

noahdietz opened the pull request:
This updates the version of Showcase used for generation in CI to v0.12.0 and adds a test for the new SequenceService to assert the basic retry attempt logic and timeout-backoff behavior works as implemented.

noahdietz commented:
I'm not really sure why the ruby tests are failing with `TypeError: Sequence is not a module` - my hunch is that the service is named `SequenceService` and there is a message named `Sequence` and somehow this results in a naming conflict.

Not really sure what to do about that. We've migrated everything to gapic-generator-ruby, maybe we shouldn't make the tests required?

vam-google commented on showcase/java/src/test/java/com/google/api/showcase/ShowcaseTest.java:
```diff
@@ -258,4 +277,66 @@ public void block() {
+    assertThat(report.getAttemptsList()).isNotNull();
+    assertThat(report.getAttemptsList().size()).isEqualTo(3);
+    for (int i = 0; i < report.getAttemptsList().size(); i++) {
+      if (i == 0) {
```
Why not just `int i = 1` in the for loop declaration?

noahdietz replied:
Good call, changed.

vam-google commented on showcase/java/src/test/java/com/google/api/showcase/ShowcaseTest.java:
```diff
@@ -258,4 +277,66 @@ public void block() {
   }
+
+  @Test
+  public void attemptSequence() {
```
The other test is `TimeoutBackoff`, can you be more specific about what this test tests?

noahdietz replied:
I opted to remove this test because it wasn't doing anything more than the `TimeoutBackoff` one. It was the first one I added as I was developing it and I removed some others that were irrelevant to this PR, but neglected to remove this one.

noahdietz commented:
Just spoke to dazuma and he said that naming collisions are definitely possible, real cases in cloud libs were worked around with synthtool, and were ultimately fixed with the microgenerator. I guess I should do something to exclude this proto from Ruby generation in this CI.

noahdietz commented:
Fixed it by breaking out ruby generation, and before generating it, I moved the offending proto out of the target directory and remove the service from the 1P yaml. Hacky, but really the only way to exclude this proto from ruby showcase gen.
